src/refinement/triquadrisection.jl

```diff
@@ -0,0 +1,66 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    TriQuadrisection()
+
+Refinement of a mesh by preliminarly triangulating it if needed and
+then subdividing each triangle into four triangles.
+
+## References
+
+* Charles Loop. 1987. [Smooth subdivision surfaces based on 
+  triangles](https://charlesloop.com/thesis.pdf). 
+  Master's thesis, University of Utah.
+"""
+struct TriQuadrisection <: RefinementMethod end
+
+function refine(mesh, ::TriQuadrisection)
+  if !(eltype(mesh) <: Triangle)
+    mesh = simplexify(mesh)
+  end
+
+  # retrieve geometry and topology
+  points = vertices(mesh)
+  npoints = length(points)
+  connec = topology(mesh)
+
+  # convert to half-edge structure
+  t = convert(HalfEdgeTopology, connec)
+  ntriangles = nelements(t)
+
+  # add middle points of edges
+  middles = Dict{Tuple{Int,Int},Int}()
+  ∂₁₀ = Boundary{1,0}(t)
+  for edge_index in 1:nfacets(t)
```

Comment:
rename edge_index to eind, or edge, or ind, ... some name that doesn't have underscores